### PR TITLE
feat: implements parallelizable transpose logic for RowMajorMatrix

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

File: matrix/src/lib.rs [hunk 1]
```diff
@@ -82,3 +82,9 @@ pub trait MatrixRowSlices<T>: MatrixRows<T> {
 pub trait MatrixRowSlicesMut<T>: MatrixRowSlices<T> {
     fn row_slice_mut(&mut self, r: usize) -> &mut [T];
 }
+
+/// A `TransposeMatrix` which supports transpose logic for matrices
+pub trait MatrixTranspose<T>: MatrixRows<T> {
+    const BLOCK_SIZE: usize;
```
Comment (dlubarov) [inline matrix/src/lib.rs:88]: Could we make this a private constant in `dense.rs`? It seems like more of an implementation detail to me

Comment (dlubarov) [reply matrix/src/lib.rs:88]: Oh I see, the `match` couldn't be used if it's a global constant. I think we could also have a `block_size` local variable and const propagation should hopefully result in the same code.

File: matrix/src/dense.rs [hunk 2]
```diff
@@ -383,3 +384,147 @@ impl<T: Clone> MatrixRowSlicesMut<T> for RowMajorMatrixViewMut<'_, T> {
...
+        16 => 16,
+        _ => 16,
+    };
+
+    fn transpose(self) -> Self {
+        let block_size = Self::BLOCK_SIZE;
+        let height = self.height();
+        let width = self.width();
+
+        let transposed_values: Vec<T> = (0..width * height).map(|_| T::default()).collect();
```
Comment (dlubarov) [inline matrix/src/dense.rs:404]: nit: could this be `vec![T::default(); width * height]`?

**Review by dlubarov — APPROVED** [review]
LGTM otherwise, as a baseline impl to improve on later

File: matrix/src/dense.rs (hunk 2 above)
Comment (jorgeantonio21) [reply matrix/src/dense.rs:404]: Yup, much better !

File: matrix/src/lib.rs (hunk 1 above)
Comment (jorgeantonio21) [reply matrix/src/lib.rs:88]: I refactored the code accordingly. Let me know your thoughts. Thanks for the suggestions !

**Comment by jorgeantonio21** [comment]
Thanks for the suggestions ! Addressed the PR comments.

**Review by dlubarov — APPROVED** [review]

**Pull request merged**